rskj-core/src/jmh/java/co/rsk/jmh/web3/plan/EthPlan.java

```diff
@@ -0,0 +1,126 @@
+
+            transactionsContractCall = TransactionFactory.createTransactions(TransactionFactory.TransactionType.CONTRACT_CALL, configuration, nonce, numOfTransactions).listIterator();
+        }
+
+
+        blockNumber = web3Connector.ethBlockNumber();
+    }
+
+    private org.web3j.protocol.core.methods.response.Transaction setupContractCallTransaction(Transaction transaction) throws IOException {
+        EthSendTransaction sendTransactionResponse = rskModuleWeb3j.ethSendTransaction(transaction).send();
+        EthTransaction transactionResponse = rskModuleWeb3j.ethGetTransactionByHash(sendTransactionResponse.getResult()).send();
+        return transactionResponse.getResult();
+    }
+
+    private org.web3j.protocol.core.methods.response.Transaction setupContractCallTransaction() throws IOException {
+        EthTransaction transactionResponse = rskModuleWeb3j.ethGetTransactionByHash(configuration.getString("eth.transactionContractCallHash")).send();
+        return transactionResponse.getResult();
+    }
+
+    @TearDown(Level.Trial) // move to "Level.Iteration" in case we set a batch size at some point
+    public void tearDown() throws InterruptedException {
+        // wait for new blocks to have free account slots for transaction creation
+        BenchmarkHelper.waitForBlocks(configuration);
+    }
+
+    public Iterator<Transaction> getTransactionsContractCall() {
+        return transactionsContractCall;
+    }
+
+    public org.web3j.protocol.core.methods.response.Transaction getContractCallTransaction() {
+        try {
+            if (contractCallTransaction == null) {
+                if (transactionsContractCall == null) {
+                    contractCallTransaction = setupContractCallTransaction();
+                } else {
+                    contractCallTransaction = setupContractCallTransaction(transactionsContractCall.next());
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return contractCallTransaction;
+    }
+
+    public RskModuleWeb3j.EthCallArguments getEthCallArguments() {
+        RskModuleWeb3j.EthCallArguments args = new RskModuleWeb3j.EthCallArguments();
+        org.web3j.protocol.core.methods.response.Transaction contractCallTransaction = getContractCallTransaction();
+
+        args.setFrom(contractCallTransaction.getFrom());
+        args.setTo(contractCallTransaction.getTo());
+        args.setGas("0x" + contractCallTransaction.getGas().toString(16));
+        args.setGasPrice("0x" + contractCallTransaction.getGasPrice().toString(16));
+        args.setValue("0x" + contractCallTransaction.getValue().toString(16));
+        args.setNonce("0x" + contractCallTransaction.getNonce().toString(16));
+        args.setChainId("0x" + BigInteger.valueOf(contractCallTransaction.getChainId()).toString(16));
+        args.setData(contractCallTransaction.getInput());
+
+        return args;
```

Comment:
can we prepare this stuff in the `setUp` method? just to avoid any possible impact on the benchmarking results..

Reply:
That is one of the main reasons of having setups 😉

Reply:
Yes, this why we have it 😉